Rewarded.swift

```diff
@@ -2,42 +2,45 @@ import SwiftUI
 import GoogleMobileAds
 import UIKit
     
-final class Rewarded: NSObject, GADRewardedAdDelegate{
-    
-    var rewardedAd:GADRewardedAd = GADRewardedAd(adUnitID: rewardID)
+
+//In v8 of Google Ads SDK GADFullScreenContentDelegate combines some functionality of GADInterstitialDelegate and GADRewardedAdDelegate
+final class Rewarded: NSObject, GADFullScreenContentDelegate {
     
-    var rewardFunction: (() -> Void)? = nil
+    var rewardedAd: GADRewardedAd?
     
     override init() {
         super.init()
         LoadRewarded()
     }
     
-    func LoadRewarded(){
-        let req = GADRequest()
-        self.rewardedAd.load(req)
-    }
-    
-    func showAd(rewardFunction: @escaping () -> Void){
-        if self.rewardedAd.isReady{
-            self.rewardFunction = rewardFunction
-            let root = UIApplication.shared.windows.first?.rootViewController
-            self.rewardedAd.present(fromRootViewController: root!, delegate: self)
+    let req = GADRequest()
+        GADRewardedAd.load(withAdUnitID: Constants.shared.rewardAdUnitId, request: req) { rewardedAd, err in
+            if let err = err {
+                print("Failed to load ad with error: \(err)")
+                return
+            }
+            
+            self.rewardedAd = rewardedAd
+            self.rewardedAd?.fullScreenContentDelegate = self
         }
-       else{
-           print("Not Ready")
```

Comment:
Why print ?

Reply:
Printing is useful for debugging. You can handle the ad not being loaded however you choose.